lib/random-util.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const RANDOM_POOL_SIZE = 8192;
+const RANDOM_POOL_REFRESH = 1024;
+
+let randomPool = Buffer.alloc(RANDOM_POOL_SIZE);
+let randomPoolIdx = RANDOM_POOL_SIZE;
+
+function onRandomBytes (err, buf) {
+  randomPending = false;
+  if (!err) {
+    randomPool = buf;
+    randomPoolIdx = 0;
+  }
+}
+
+let randomPending = true;
+crypto.randomBytes(RANDOM_POOL_SIZE, onRandomBytes);
+
+function randomFillMask (buffer) {
+  if (RANDOM_POOL_SIZE - randomPoolIdx < 4) {
+    return crypto.randomFillSync(buffer, offset, size);
```

Comment:
It's still best to try to refresh the whole poll at this point also

Reply:
BTW once this if triggers it will trigger forever. Granted it won't trigger in most of the cases .. you have plenty of room but yet it's dangerous as it is

Reply:
Why do you thing it will trigger forever? When we are here then we are already refreshing the pool. So this will only trigger until the randomBytes callback has been invoked.

Reply:
Uh yes .. you're right this is just the  specifics of the side effects of this particular case. Just put a comment here to help the next one reading the code. simply returning like this without any state change is huge red light in most heads. BTW randomPending might  go as well?